presto-main/src/main/java/com/facebook/presto/sql/planner/SortExpressionExtractor.java

```diff
@@ -82,71 +89,104 @@ private SortExpressionExtractor() {}
     private static SortExpressionContext merge(SortExpressionContext left, SortExpressionContext right)
     {
         checkArgument(left.getSortExpression().equals(right.getSortExpression()));
-        ImmutableList.Builder<Expression> searchExpressions = ImmutableList.builder();
+        ImmutableList.Builder<RowExpression> searchExpressions = ImmutableList.builder();
         searchExpressions.addAll(left.getSearchExpressions());
         searchExpressions.addAll(right.getSearchExpressions());
         return new SortExpressionContext(left.getSortExpression(), searchExpressions.build());
     }
 
     private static class SortExpressionVisitor
-            extends AstVisitor<Optional<SortExpressionContext>, Void>
+            implements RowExpressionVisitor<Optional<SortExpressionContext>, Void>
     {
         private final Set<Symbol> buildSymbols;
+        private final FunctionManager functionManager;
 
-        public SortExpressionVisitor(Set<Symbol> buildSymbols)
+        public SortExpressionVisitor(Set<Symbol> buildSymbols, FunctionManager functionManager)
         {
             this.buildSymbols = buildSymbols;
+            this.functionManager = functionManager;
         }
 
         @Override
-        protected Optional<SortExpressionContext> visitExpression(Expression expression, Void context)
+        public Optional<SortExpressionContext> visitCall(CallExpression call, Void context)
         {
-            return Optional.empty();
-        }
+            FunctionMetadata functionMetadata = functionManager.getFunctionMetadata(call.getFunctionHandle());
```

Comment:
nit: use FunctionResolution?

Reply:
You still need `FunctionMetadata` from `FunctionManager`, aren't ya

Reply:
Yeah, we cannot get operatorType from resolution anyway ...